SerialPrograms/Source/PokemonHome/Programs/PokemonHome_NationalDexSorter.cpp

```diff
@@ -0,0 +1,1532 @@
+            "Sort available " + STRING_POKEMON + " according to the National Dex number.",
+            FeedbackType::REQUIRED, true,
+            PABotBaseLevel::PABOTBASE_12KB
+        )
+    {}
+
+
+
+    NationalDexSorter::NationalDexSorter()
+        : START_FROM_DEX_NO(
+                  "<b>Pokédex # of first Pokémon to sort:</b>",
+                  1
+        )
+        , GET_TO_DEX_NO(
+            "<b>Pokédex # of last Pokémon to sort:</b>",
+            905
+        )
+        , START_FROM_PAGE(
+            "<b>Box Page to use for the sorted Pokémon:</b>",
+            5
+        )
+        , CHOOSE_SHINY_FIRST(
+            "<b>Look for shiny Pokémon first:</b>",
+            true
+        )
+    {
+        PA_ADD_OPTION(START_LOCATION);
+        PA_ADD_OPTION(START_FROM_DEX_NO);
+        PA_ADD_OPTION(GET_TO_DEX_NO);
+        PA_ADD_OPTION(START_FROM_PAGE);
+        PA_ADD_OPTION(CHOOSE_SHINY_FIRST);
+    }
+
+    int descriptorStartDexNo = 1;
+    int descriptorEndDexNo = 905;
+    int descriptorStartPage = 5;       // first box to populate
+    const int leSpeed = 11;                 // ticks to wait after most of the button presses, seems like this is good enough for most presses.
+    const double lineWidth = 0.024;         // how big is the width of the line we're going to check (it works but I have no idea how big this actually is)
+    const uint lineCol1 =0xf44336;          // the red to check for in the name bar
+    const uint lineCol2 =0xf44336;          // the red to check for in the name bar
+    int eCount = 0;                         // if at any point you make too many moves before getting to the next step give up on the pokémon
+    const int eCountMax = 7;               // this is the max steps I allow before giving up on a given task
+
+PokemonLA::ItemCompatibility detect_item_compatibility(const ImageViewRGB32& screen,const ImageFloatBox& box,const uint& col1,const uint& col2,const int& superCheck){
+    // This is copied from the SkipToFullMoon code, I use it as a solid color check.
+    // I use it in various ways in order to reach my goals, not all of them make sense, but all work.
+
+    const bool replace_background = true;
+    ImageRGB32 region = filter_rgb32_range(
+        extract_box_reference(screen, box),
+        col1, col2, Color(0), replace_background
+    );
+
+    ImageStats stats = image_stats(region);
+    // std::cout << "Compability color " << stats.average.r << " " << stats.average.g << " " << stats.average.b << std::endl;
+    if (stats.average.r > stats.average.b + 50.0){
+        return PokemonLA::ItemCompatibility::COMPATIBLE;
+    }
+    if (stats.average.b > stats.average.r + 50.0){
+        return PokemonLA::ItemCompatibility::INCOMPATIBLE;
+    }
+    //I created these 2 checks for more specific colors, no idea how it works, just know it works :P
+    if (stats.average.b > 254.5 && stats.average.r > 252.5 && superCheck == 1){
```

Comment:
the check with == 1 is never used. Could the check with ==2 be used in all cases ? having "bool parameter" isn't great usually

Reply:
Got rid of the variable altogether.